### PR TITLE
SpeakOnUseSystem and phrases for figurines

### PR DESCRIPTION
**Pull request by lzk228**
## About the PR

Add system and component that allows an entity to say strings from LocalizedDatasetPrototype, works mostly like OnUiClose in vending machines 

## Why / Balance

Funny feature

## Technical details

Nuh

## Media

https://github.com/user-attachments/assets/f2dcfbe1-a62b-4af5-b3d1-bdb69b1f1a18

- [X] I have added screenshots/videos to this PR showcasing its changes ingame, **or** this PR does not require an ingame showcase

## Breaking changes

**Changelog**

:cl:
- add: Figurines will say phrases on activation.

### CODE REVIEW

**Comment by lzk228** [comment]
Ready for review

File: Content.Server/Chat/Systems/SpeakOnUseSystem.cs [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+    [Dependency] private readonly IPrototypeManager _prototypeManager = default!;
+    [Dependency] private readonly ChatSystem _chat = default!;
+
+    public override void Initialize()
+    {
+        base.Initialize();
+        SubscribeLocalEvent<SpeakOnUseComponent, UseInHandEvent>(TrySpeak);
+    }
+
+    public void TrySpeak(EntityUid uid, SpeakOnUseComponent? component, UseInHandEvent args)
```
Comment (slarticodefast) [inline Content.Server/Chat/Systems/SpeakOnUseSystem.cs:26]: Should be named `OnUseInHand` as far as I understand the naming conventions.

File: Resources/Locale/en-US/datasets/figurines.ftl [hunk 2]
```diff
@@ -0,0 +1,180 @@
+figurines-hop-1 = Papers, please.
+figurines-hop-2 = You are fired.
+
+figurines-passenger-1 = How to change hands?
+figurines-passenger-2 = Call the evac.
```
Comment (slarticodefast) [inline Resources/Locale/en-US/datasets/figurines.ftl:5]: Should it be `call the evac` or just `call evac` instead?

File: Resources/Locale/en-US/datasets/figurines.ftl [hunk 3]
```diff
@@ -0,0 +1,180 @@
...
+figurines-qm-2 = I won't approve the guns.
+figurines-qm-3 = I didn't buy those guns!
+figurines-qm-4 = One toys crate for ma fellow clown!
+figurines-qm-5 = Time to spent all money on gambling.
+figurines-qm-6 = Viva La Cargonia!
+figurines-qm-7 = Fill the form.
+
+figurines-ce-1 = Everyone to the briefing!
+figurines-ce-2 = Wire the solars!
+figurines-ce-4 = How to setup the TEG?
```
Comment (slarticodefast) [inline Resources/Locale/en-US/datasets/figurines.ftl:64]: Number 3 is missing and causing your tests to fail.

File: Resources/Locale/en-US/datasets/figurines.ftl [hunk 4]
```diff
@@ -0,0 +1,180 @@
...
+figurines-ce-5 = SINGULOOSE!
+figurines-ce-6 = TESLALOOSE!
```
Comment (slarticodefast) [inline Resources/Locale/en-US/datasets/figurines.ftl:66]: According to my highly accurate opinion poll (i.e. used the search function on discord) it should be `TESLOOSE!`
grafik
grafik

File: Resources/Locale/en-US/datasets/figurines.ftl [hunk 5]
```diff
@@ -0,0 +1,180 @@
...
+
+figurines-engineer-1 = SINGULOOSE!
+figurines-engineer-2 = TESLALOOSE!
```
Comment (slarticodefast) [inline Resources/Locale/en-US/datasets/figurines.ftl:69]: same

File: Resources/Locale/en-US/datasets/figurines.ftl [hunk 6]
```diff
@@ -0,0 +1,180 @@
...
+
+figurines-footsoldier-1 = I'm an evil boy. Less boy every day, more evil every day.
+figurines-footsoldier-2 = Who will you choose? Them or us? Us or them?
+
+figurines-wizard-1 = Ei Nath!
+figurines-wizard-2 = Wehgardium Leviosa!
+figurines-wizard-3 = Skidaddle skadoodle!
+figurines-wizard-4 = FIREBALL!
+
+figurines-space-dragon-1 = Fish will consume the staion.
```
Comment (slarticodefast) [inline Resources/Locale/en-US/datasets/figurines.ftl:136]: station

File: Resources/Locale/en-US/datasets/figurines.ftl [hunk 7]
```diff
@@ -0,0 +1,180 @@
...
+figurines-hamlet-1 = Piep!
+figurines-hamlet-2 = Squeak!
+figurines-hamlet-3 = Chuu!
+figurines-hamlet-4 = Eeee!
+figurines-hamlet-5 = Pip!
+figurines-hamlet-6 = Fwiep!
+figurines-hamlet-7 = Heep!
+figurines-hamlet-8 = NOT THE MICROWAVE!
+
+
```
Comment (slarticodefast) [inline Resources/Locale/en-US/datasets/figurines.ftl:168]: remove the empty lines below

**Review by slarticodefast — COMMENTED** [review]
I absolutely love the idea.
Tested it in-game and works as intended.

File: Resources/Prototypes/Datasets/figurines.yml [hunk 8]
```diff
@@ -0,0 +1,263 @@
...
+  id: FigurinesQM
+  values:
+    prefix: figurines-qm-
+    count: 7
+
+- type: localizedDataset
+  id: FigurinesCE
+  values:
+    prefix: figurines-ce-
+    count: 6
```
Comment (slarticodefast) [inline Resources/Prototypes/Datasets/figurines.yml:107]:
```suggestion
    count: 5
```
Should fix the integration test failing

Comment (lzk228) [reply Resources/Prototypes/Datasets/figurines.yml:107]: oh right, i forgor

**Comment by Agustron24** [comment]
Make commander figurine say "my edibles kicked in at a bad time sorry"

**Pull request merged**